### ISSUE TITLE
5318 w2

### ISSUE THREAD

**Issue by henryliangt** [henryliangt/usyd#5]
The difference between all distance algorithms?
What use case will hv different result?
Most case seem result ?

**Comment by henryliangt** [comment]
Assume 1 million data.

K-nearest neighbor commercial package typically use 10.
Even number votes ?
More k samples to see the consistency ?

**Comment by henryliangt** [comment]
Weighted nearest neighbor w=1/d*d

**Comment by henryliangt** [comment]
Decision boundaries with an arbitrary shape formed by Voronoi diagram

**Comment by henryliangt** [comment]
w3
how to calculate the b0 b1,
regression coefficient.

**Comment by henryliangt** [comment]
closest line to the data,
mathematically prediction error



performance index = sum of squared prediction errors (SSE) 

求解 method of the least squares.最小二乘法。

**Comment by henryliangt** [comment]
最小二乘法，
微积分  differential calculus 
附录。

**Comment by henryliangt** [comment]
coefficient of determination Fit

最小二乘法求解后 SSE = 12 ，评估?

R^2 = SSR / SST  (0, 1) 越大越好。



SSE Sum of squared prediction errors,
actual - predicted


SST  Sum of squared total errors,
actual - mean 

is a function of the variance of y(variance = standard deviation ^ 2) => 
SST is a measure of the variability of y, without considering x,

can  b  baseline,最差的基本情况。



SSR sum of squared regression errors = predicted value - mean value,

**Comment by henryliangt** [comment]
differential calculus, residual, sigmoid curve , logistic curve , simple (bivariate) logistic regression ,

logistic curve

maximum likelihood method 
最大似然法find the parameters b0 and b1, the curve best fit the data.

**Comment by henryliangt** [comment]
logistic regression equation

appendix 2 

linear calculation as in linear regression.


called odds ratio for the default class (class 1)

**Comment by henryliangt** [comment]
Ridge regression  岭回归
Lasso regression   套索回归
Logistic regression 逻辑回归

**Comment by henryliangt** [comment]
Some algorithms are more susceptible to overfitting than others  易受影响的
Decision tree : prune the tree  修剪树
Neural networks : early stopping of the training

**Comment by henryliangt** [comment]
sweet spot : point in between, which will yield the best test accuracy

**Comment by henryliangt** [comment]
Regularization: means explicitly restricting a model to avoid overfitting. (Ridge and Lasso regression
+ nn)

**Comment by henryliangt** [comment]
### Ridge regression
A regularized version of the standard Linear Regression (LR) 
• Also called Tikhonov regularization
• Uses the same equation as LR to make predictions
• However, the regression coefficients w are chosen so that they not only 
fit well the training data (as in LR) but also satisfy an additional 
constraint: 
• the magnitude of the coefficients is as small as possible, i.e. close 
to 0 
• Small values of the coefficients means
• each feature will have little effect on the outcome 
• small slope of the regression line
• Rationale: a more restricted model (less complex) is less likely to overfit
• Ridge regression uses the so called L2 regularization (L2 norm of the 
weight vector)

**Comment by henryliangt** [comment]
### Ridge regression (2)
Minimizes the following cost function:
• Parameter  controls the trade-off between the performance on 
training set and model complexity 
1
𝑛
σ𝑖=1
𝑛
(𝑦ො𝑖 − 𝑦𝑖ሻ
2+ 𝛼 σ𝑖=1
𝑚 𝑤𝑖
2
MSE regularization term
Goal: high accuracy 
on training data (low 
MSE)
low complexity 
model – w close to 0

**Comment by henryliangt** [comment]
### Lasso regression
##### Least Absolute Shrinkage and Selection Operator Regression

Another regularized version of the standard Linear Regression (LR) 
 As Ridge Regression, it adds a regularization term to the cost function but it uses the L1 norm of the regression coefficient vector w 

• Consequence of using L1 – some w will become exactly 0
• => some features will be completely ignored by the model – a form of automatic feature selection
• Less features – simpler model, easier to interpret

**Comment by henryliangt** [comment]
### Minimizing SSE App 1
D. Larose, Data Mining: Methods and Models, 2006, Wiley; p.36-37

![image](https://user-images.githubusercontent.com/23263731/187069102-4832db06-61d8-44f7-82e5-917c209395b2.png)

![image](https://user-images.githubusercontent.com/23263731/187069107-5a8374a4-1fbe-45ae-b913-565274d72628.png)

### Appendix 2: Logistic regression
![image](https://user-images.githubusercontent.com/23263731/187069127-a772825d-7a9d-4a62-b0ea-f41a6d4dc951.png)

**Comment by henryliangt** [comment]
bivariate   双变量
hyperplane    超平面

**Comment by henryliangt** [comment]
Pang-Ning Tan, Michael Steinbach, Anuj Karpathe and Vipin Kumar (2019). Introduction to Data Mining, 2nd edition. Pearson. (You can also use the previous edition)

Ian H. Witten, Eibe Frank, Mark Hall and Christopher J. Pal

Data Mining - Practical Machine Learning Tools and Techniques, 4th edition, Morgan Kaufmann, 2017 (You can also use the 3nd edition)


Andreas C. Mueller and Sarah Guido (2016). Introduction to Machine Learning with Python: a Guide for Data Scientists, O'Reilly

Aurelien Geron (2019). Hands-On Machine Learning with Scikit-Learn, Keras and TensorFlow, O'Reilly.
